src/random.h

```diff
@@ -119,14 +119,14 @@ void RandAddEvent(const uint32_t event_info) noexcept;
 class FastRandomContext
 {
 private:
-    bool requires_seed;
-    ChaCha20 rng;
+    bool requires_seed{true};
+    ChaCha20 rng{};
 
-    unsigned char bytebuf[64];
-    int bytebuf_size;
+    unsigned char bytebuf[64] = {};
```

Comment:
shouldn't this be left uninitialized? With this patch memory sanitizer will fail to detect logic errors.